internalserver/handler.go

```diff
@@ -0,0 +1,104 @@
+package internalserver
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"sort"
+
+	"github.com/metalmatze/signal/healthcheck"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Handler struct {
+	http.ServeMux
+	endpoints map[string]string
+}
+
+func NewHandler(options ...Option) *Handler {
+	h := &Handler{endpoints: map[string]string{}}
+
+	h.HandleFunc("/", h.index)
+
+	for _, option := range options {
+		option(h)
+	}
+
+	return h
+}
+
+func (h *Handler) AddEndpoint(pattern string, description string, handler http.HandlerFunc) {
+	h.endpoints[pattern] = description
+	h.HandleFunc(pattern, handler)
+}
+
+func (h *Handler) index(w http.ResponseWriter, r *http.Request) {
+	html := "<html><head><title>Internal</title></head><body>\n"
+
+	// No follows some sorting of endpoints to always show them in the same order.
```

Comment:
What do we mean by `no follows`?

Reply:
Now?

Reply:
Yes, that's a typo. I'll update. It should be something like: "On the next couple of lies we are sorting the endpoints to always show them in the same order."